common/channel/src/message_queues.rs

```diff
@@ -139,18 +154,51 @@ impl<K: Eq + Hash + Clone, T> PerKeyQueue<K, T> {
                 return None;
             }
         };
+
         let (message, is_q_empty) = self.pop_from_key_queue(&key);
         if !is_q_empty {
             self.round_robin_queue.push_back(key);
         }
+
         if message.is_some() {
             if let Some(c) = self.counters.as_ref() {
                 c.with_label_values(&["dequeued"]).inc();
             }
+
+            // Remove empty per-key-queues every `POPS_PER_GC` successful dequeue
+            // operations.
+            //
```

Comment:
We'll probably want GC based on time or something else if we find this to be expensive, but this is good for now!

Reply:
my initial intuition was to make `POPS_PER_GC` much larger (like 1000+), but this seemed to be slower for whatever reason (of course, this is just in a small micro benchmark, so take everything with several grains of salt)

Reply:
Weird, we'll probably later have to test it on cluster test or something with a larger scale